Cython/Compiler/Builtin.py

```diff
@@ -110,6 +113,53 @@ def declare_in_type(self, self_type):
         )
 
 
+def _generate_divmod_function(scope, argument_types):
```

Comment:
In principle it might be possible to treat extern and non-extern typedefs differently. In that non-extern typedefs we know exactly which integer type they correspond to, so those can just use the known integer overload